src/function/vector_cast_functions.cpp

```diff
@@ -174,10 +174,20 @@ static bool hasImplicitCastStruct(const LogicalType& srcType, const LogicalType&
     return true;
 }
 
-static bool hasImplicitCastUnion(const LogicalType& /*srcType*/, const LogicalType& /*dstType*/) {
-    // todo: implement union casting function
-    // currently, there seems to be no casting functionality between union types
-    return false;
+static bool hasImplicitCastUnion(const LogicalType& srcType, const LogicalType& dstType) {
+    // srcType is either non-nested or a union
+    if (srcType.getLogicalTypeID() == LogicalTypeID::UNION) {
```

Comment:
I guess, we also don't support nested type cast to union in this PR right?
We should also throw an exception in such case.